File: code-examples/php/usage-examples/distinct.php
<?php
require 'vendor/autoload.php'; // Include the MongoDB SDK

use MongoDB\Client as MongoClient; // Import the MongoDB classes required

$uri = 'mongodb://localhost:27017'; // replace the uri string with your MongoDB deployment's connection string.

$collection = (new MongoClient($uri))->sample_mflix->movies;

$filter = ['directors' => 'Barbra Streisand'];

try {
    $result = $collection->distinct('year', $filter);
    print_r($result);

} catch (Exception $e) {
    echo $e->getMessage(), "\n";
}
?>

File: code-examples/scala/usage-examples/distinct.scala
import org.mongodb.scala._

// Replace the uri string with your MongoDB deployment's connection string.
val uri = "<connection string uri>"

val mongoClient: MongoClient = MongoClient(uri)

val database: MongoDatabase = mongoClient.getDatabase("sample_mflix")

val movies: MongoCollection[Document] = database.getCollection("movies")

val observable = movies.distinct("year", classOf[Int], Filters.eq("directors", "Barbra Streisand"))

// Attach an observer
observable.subscribe(new Observer[Int] {
  override def onNext(year: Int): Unit = println(year)
  override def onError(e: Throwable): Unit = e.printStackTrace()
  override def onComplete(): Unit = println("Operation completed")
})

// Don't forget to close the connection
mongoClient.close()
